.swm/state-management-with-reflux.jni2e.sw.md

```diff
@@ -0,0 +1,193 @@
+---
+id: jni2e
+name: State Management with Reflux
+file_version: 1.0.2
+app_version: 0.8.5-0
+file_blobs:
+  static/app/actions/pageFiltersActions.tsx: 8d6f2485c157b80586df789ae5c655bf51299ec7
+  static/app/stores/pageFiltersStore.tsx: c04a27b5e33fd62b865b58b4bddac1be1b899f3e
+  static/app/actionCreators/pageFilters.tsx: 2b27c6d727455f7dde4fa3ea8210a4b6162b661f
+  static/app/types/core.tsx: c27f92c56ab334744f9a5a39990b7b86e6c1aa27
+---
+
+We currently use [Reflux](https://github.com/reflux/refluxjs) for managing global state. Reflux is a simple library for unidirectional dataflow architecture inspired by ReactJS [Flux](http://facebook.github.io/react/blog/2014/05/06/flux.html).
+
+# Overview
+
+The main function of Reflux is to introduce a more functional programming style architecture by adopting a single data flow pattern.
+
+```
++---------+       +--------+       +-----------------+
+¦ Actions ¦------>¦ Stores ¦------>¦ View Components ¦
++---------+       +--------+       +-----------------+
+     ^                                      ¦
+     +--------------------------------------+
+```
+
+The pattern is composed of actions and data stores, where actions initiate new data to pass through data stores before coming back to the view components again. If a view component has an event that needs to make a change in the application's data stores, they need to do so by signaling to the stores through the actions available.
```

Comment:
In this pattern, actions initiate the process of data passing through data stores before returning to the view components. 

I feel like this is a bit clearer, wdyt?

Reply:
A view component's event that requires changes to the application's data stores signals the stores with the available actions.